models/contacts.js

```diff
@@ -1,29 +1,112 @@
-const Contact = require("../models/contactSchema");
+const express = require("express");
+const router = express.Router();
+const Joi = require("joi");
+const authMiddleware = require("../middlewares/authMiddleware");
+const {
+  listContacts,
+  getContactById,
+  removeContact,
+  addContact,
+  updateContact,
+} = require("../models/contacts");
 
-const listContacts = async () => {
-  return await Contact.find();
-};
+const contactSchema = Joi.object({
+  name: Joi.string().required(),
+  email: Joi.string().email().required(),
+  phone: Joi.string().required(),
+});
 
-const getContactById = async (contactId) => {
-  return await Contact.findById(contactId);
-};
+const favoriteSchema = Joi.object({
+  favorite: Joi.boolean().required(),
+});
 
-const removeContact = async (contactId) => {
-  return await Contact.findByIdAndRemove(contactId);
-};
+router.get("/", authMiddleware, async (req, res, next) => {
+  try {
+    const contacts = await listContacts({ owner: req.user._id });
+    res.status(200).json(contacts);
+  } catch (error) {
+    next(error);
+  }
+});
 
-const addContact = async (body) => {
-  return await Contact.create(body);
-};
+router.get("/:id", authMiddleware, async (req, res, next) => {
+  try {
+    const contact = await getContactById(req.params.id);
+    if (!contact || contact.owner.toString() !== req.user._id.toString()) {
```

Comment:
Ten komentarz dotyczy wszystkich innych ścieżek, żebyś wysyłał dwa filtry jeden No to jest klucz owner a drugi klucz To jest ID. W ten sposób zawsze masz pewność że jeżeli obydwa warunki są spełnione to zostanie wykonane działanie na bazie danych